packages/composer/amazeelabs/graphql_directives/src/Plugin/GraphQL/DataProducer/DrupalView.php

```diff
@@ -0,0 +1,136 @@
+
+  public function resolve(string $id, ?string $args, FieldContext $context): array {
+    return $this->withCacheContext($context, function() use ($id, $args, $context) {
+      [$viewId, $viewDisplayId] = explode(':', $id);
+
+      $view = Views::getView($viewId);
+      if (!$view) {
+        throw new \Exception("View {$viewId} not found.");
+      }
+
+      if (!$view->setDisplay($viewDisplayId)) {
+        throw new \Exception("Display {$viewDisplayId} not found in view {$viewId}.");
+      }
+
+      $arguments = [];
+      if ($args) {
+        parse_str($args, $arguments);
+      }
+
+      $pageSize = 10;
+      if (isset($arguments['pageSize']) && is_numeric($arguments['pageSize'])) {
+        $pageSize = (int) $arguments['pageSize'];
+      }
+      $view->setItemsPerPage($pageSize);
+      unset($arguments['pageSize']);
+
+      $page = 1;
+      if (isset($arguments['page']) && is_numeric($arguments['page'])) {
+        $page = (int) $arguments['page'];
+      }
+      $offset = ($page - 1) * $pageSize;
+      $view->setOffset($offset);
+      unset($arguments['page']);
+
+      $contextualFilters = [];
+      if (isset($arguments['contextualFilters']) && is_string($arguments['contextualFilters'])) {
+        $contextualFilters = explode('/', $arguments['contextualFilters']);
+        $view->setArguments($contextualFilters);
+      }
+      unset($arguments['contextualFilters']);
+
+      if (!empty($arguments)) {
+        $view->setExposedInput($arguments);
+      }
+
+      $view->get_total_rows = TRUE;
+      $view->execute();
+
+      $rows = array_map(function(ResultRow $row) use ($context) {
+        $entity = $row->_entity;
+        if (!($entity instanceof EntityInterface)) {
+          throw new \Exception("Row is not an entity.");
+        }
+        if ($entity instanceof TranslatableInterface && $entity->hasTranslation($context->getContextLanguage())) {
+          $entity = $entity->getTranslation($context->getContextLanguage());
+        }
+        return $entity;
+      }, $view->result);
+
+      $total = ($view->total_rows ?: 0) + $offset;
+
+      $form = $view->exposed_widgets;
+      $filters = [];
+      foreach (Element::children($form) as $key) {
+        if (isset($form[$key]['#options'])) {
+          foreach ($form[$key]['#options'] as $value => $label) {
+            $filters[$key][] = [
+              'value' => (string) $value,
+              'label' => (string) $label,
+            ];
+          }
+        }
+      }
+
+      $context->addCacheableDependency($view);
+
+      return [
+        '#info' => [
+          'view' => $view,
+          'page_size' => $pageSize,
+          'page' => $page,
+          'contextual_filters' => $contextualFilters,
+        ],
+        'rows' => $rows,
+        'total' => $total,
+        'filters' => $filters,
+      ];
+    });
+  }
+
+  function withCacheContext(FieldContext $resolveContext, callable $callback) {
```

Comment:
Sometimes I wonder if the GraphQL module should do that by default 😅

Reply:
There's an ultimate solution: https://www.drupal.org/project/drupal/issues/2638686#comment-13818152